qiime2/sdk/usage.py

```diff
@@ -0,0 +1,344 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import abc
+import re
+import types
+
+from qiime2 import sdk
+
+# TODO: docstrings
+
+
+class UsageAction:
+    def __init__(self, *, plugin_id: str, action_id: str):
```

Comment:
Do we need this to be keyword only?  I'm usually all for them, but this seems like a situation where it's pretty obvious which arg is which, and I'm not sure what extensions to this idea would ever come about.

Reply:
It forces a consistent style across all the `UsageAction/UsageInputs/UsageOutputNames` invocations, which I like:

```python
    use.action(
        UsageAction(plugin_id='dummy_plugin', action_id='typical_pipeline'),
        UsageInputs(int_sequence=ints2, mapping=mapper2, do_extra_thing=False),
        UsageOutputNames(out_map='out_map2', left='left2', right='right2',
                         left_viz='left_viz2', right_viz='right_viz2')
    )
```

vs 

```python
    use.action(
        UsageAction('dummy_plugin', typical_pipeline'),
        UsageInputs(int_sequence=ints2, mapping=mapper2, do_extra_thing=False),
        UsageOutputNames(out_map='out_map2', left='left2', right='right2',
                         left_viz='left_viz2', right_viz='right_viz2')
    )
```

I am open to dropping the keyword-only nature, though.

Reply:
That seems like a good reason